src/jrd/jrd.cpp

```diff
@@ -384,6 +384,9 @@ static void threadDetach()
 {
 	ThreadSync* thd = ThreadSync::findThread();
 	delete thd;
+
+	if (cds::threading::Manager::isThreadAttached())
```

Comment:
Please encapsulate usage of this library somewhere, like we do with others external libraries.

Reply:
Could you be more specific here, what exactly do you mean ?

Reply:
We have encapsulation of extern code: ICU (unicode_utils), tommath (BigInteger), tomcrypt (TomCryptHash). absl (Int128).

Why use this library here directly in `jrd.cpp` instead of following this good practice?

Reply:
I still not undertand what you want. Callbacks from engine (?) to attach\detach threads ?
I see no big sence in implementing it, at least until we have just one library that requires it.
Something else ? Explain, pls

BTW, why some methods of library is OK to call from engine and others - not OK ?